src/Components/Admin/EnvironmentEditor/environment.js

```diff
@@ -101,11 +103,32 @@ export default function EnvironmentEditor({
         setScenes(items);
     };
 
+    const onCreateButtonClick = () => {
+        setCreateModalOpen(true);
+    };
+
+    const onCreateModalClose = () => {
+        setCreateModalOpen(false);
+    };
+
+    const onCreateModalSubmit = async (name, background_id) => {
+        setCreateModalOpen(false);
+
+        const newScene = await createScene(name, background_id);
+        const newSceneData = [...scenes, newScene.id];
+        setScenes(newSceneData);
+
+        const newEnvData = environment;
+        newEnvData.scene_ids = [...environment.scene_ids, newScene.id.id];
+        const newEnv = await editScenario(newEnvData);
+        setEnvironment(newEnv);
```

Comment:
This is more complicated than it needs to be. I propose that we change our backend scene POST endpoint to include the environment ID so that the environment is automatically updated rather than having to send two requests.

Furthermore, the response that I get from scene POST endpoint is a little weird, hence why I used `newScene.id`.  Not sure why the data is nested under `id`. 

<img width="514" alt="Screen Shot 2021-05-03 at 9 53 12 AM" src="https://user-images.githubusercontent.com/21224282/116899913-6254e480-abf5-11eb-9d84-09ec177154c8.png">

Thoughts on the above @ahmedhamodi @jaydhulia

Reply:
I don't think we should change the backend scene POST endpoint. The scene is meant to be separate from a scenario, so including a scenario ID with a scene creation doesn't make sense to me. Ideally, admins create scene on the scenes tab that Ahmed has created. And then on the scenario, when they want to add a scene, they simply select from already existing scenes.
That was the workflow that was kept in mind when creating the backend database schema. That makes more sense to me, because admins may want to reuse scenes across scenarios.
The scenario is simply to have an ordered collection of scenes and control whether they are published.

The other `id` issue has been addressed via Slack.

Reply:
> Furthermore, the response that I get from scene POST endpoint is a little weird, hence why I used newScene.id. Not sure why the data is nested under id.

Resolved

Reply:
I think it's fine to expose creation of a scene in both locations personally.

It's just when it's done here, the linking to an environment needs to be explicitly done.

If done through the scenes tab, that is not needed as it is independent and not linked to any environment.

Because of this second flow that doesn't need the linkage, it should be done separate from the POST endpoint.